aeron-agent/src/main/java/io/aeron/agent/ArchiveEventDissector.java

```diff
@@ -528,7 +528,9 @@ static void dissectReplaySessionStateChange(
         final long position = buffer.getLong(absoluteOffset, LITTLE_ENDIAN);
         absoluteOffset += SIZE_OF_LONG;
 
-        builder.append(": sessionId=").append(sessionId);
+        builder.append(": replaySessionId=").append(sessionId);
+        builder.append(" replayId=").append(sessionId >> 32);
+        builder.append(" sessionId=").append(sessionId & 0xFFFF_FFFFL);
```

Comment:
You can just cast to `int`.

Reply:
Is it 'better' to cast it to an int?  I don't have strong feelings either way, but I kind of like the explicit nature of doing a bitwise AND.

Reply:
Casting to an int yields a different value compared with '& 0xFFFF_FFFFL'.  The sign gets switched.

Reply:
Oh I get it... if I use the &, the returned value is still a long, and the last bit is 0, so the number is positive.  If I cast to an int, it's an int... in which case (in the case of 20000000000), the last bit is a 1.  So the number is negative.  The 32 bits in question are the same, but how it looks when it gets printed is quite different.

Which is 'correct'?  Seems like in most places the session id is passed around as a long, in which case casting to an int and printing that is probably the wrong approach.

Reply:
`Publication#sessionId` is a 32 bit value (`int` in Java) so casting to an `int` is correct way to extract this value out of `long` value.
Running this test:
```java
@ParameterizedTest
    @ValueSource(ints = { Integer.MIN_VALUE, -777, -1, 0, 1, 42, Integer.MAX_VALUE })
    void extractSessionId(final int sessionId)
    {
        final int replayId = 0xFF;
        final long replicationSessionId = ((long)replayId << 32) | sessionId;
        System.out.println("replicationSessionId=" + replicationSessionId);
        System.out.println("cast: " + (int)replicationSessionId);
        System.out.println("mask: " + (replicationSessionId & 0xFFFF_FFFFL));
    }
```
I get the following output:
```
replicationSessionId=-2147483648
cast: -2147483648
mask: 2147483648
replicationSessionId=-777
cast: -777
mask: 4294966519
replicationSessionId=-1
cast: -1
mask: 4294967295
replicationSessionId=1095216660480
cast: 0
mask: 0
replicationSessionId=1095216660481
cast: 1
mask: 1
replicationSessionId=1095216660522
cast: 42
mask: 42
replicationSessionId=1097364144127
cast: 2147483647
mask: 2147483647
```